HRIS.Services.Tests/Services/EmployeeDocumentServiceUnitTest.cs

```diff
@@ -189,6 +190,60 @@ public async Task SaveEmployeeDocumentOnProfilePass()
 
 
 
+        List<Role> roles = new List<Role> { new Role(EmployeeRoleTestData.RoleDtoEmployee) };
+        _employeeServiceMock.Setup(x => x.GetById(employeeId))
+            .ReturnsAsync(EmployeeTestData.EmployeeDto);
+
+        _unitOfWorkMock
+            .Setup(x => x.EmployeeRole.Get(It.IsAny<Expression<Func<EmployeeRole, bool>>>()))
+            .Returns(EmployeeRoleTestData.EmployeeRolesList.AsQueryable().BuildMock());
+
+        _unitOfWorkMock
+            .Setup(x => x.Role.Get(It.IsAny<Expression<Func<Role, bool>>>()))
+            .Returns(roles.AsQueryable().BuildMock());
+
+        _unitOfWorkMock.Setup(x => x.EmployeeDocument.Add(It.IsAny<EmployeeDocument>()))
+            .ReturnsAsync(EmployeeDocumentTestData.EmployeeDocumentPending);
+
+        var result = await _employeeDocumentService.SaveEmployeeDocument(EmployeeDocumentTestData.SimpleDocumentDto, "test@retrorabbit.co.za", 3);
+
+        Assert.NotNull(result);
+        Assert.Equal(EmployeeDocumentTestData.EmployeeDocumentPending.DocumentType, result.DocumentType);
+        Assert.Equal(EmployeeDocumentTestData.EmployeeDocumentPending, result);
+        _employeeServiceMock.Verify(x => x.GetById(employeeId), Times.Once);
+        _unitOfWorkMock.Verify(x => x.EmployeeDocument.Add(It.IsAny<EmployeeDocument>()), Times.Once);
+    }
+
+    [Fact]
+    public async Task SaveAdministrativeDocumentPass()
```

Comment:
Perhaps a more descriptive name to describe why it passes could be **SaveAdministrativeDocumentPass_ShouldReturnSavedDocument**